Library/Homebrew/shims/mac/super/curl-config

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash -p
+
+# HOMEBREW_LIBRARY is set by bin/brew
+# HOMEBREW_CURL_CONFIG is set by brew.sh
+# shellcheck disable=SC2154
+source "${HOMEBREW_LIBRARY}/Homebrew/shims/utils.sh"
+
+# Check if HOMEBREW_CURL_CONFIG is set, fallback to system curl-config if not.
+curl_config_exec="${HOMEBREW_CURL_CONFIG:-/usr/bin/curl-config}"
+
+# Intercept --features and append HTTP2 if needed
+if [[ "$1" == "--features" ]]
+then
+  output="$("${curl_config_exec}" "$@")"
+
+  # Check if HTTP2 is missing but supported
+  if [[ ! "${output}" =~ HTTP2 ]]
+  then
+    curl_version="$("${HOMEBREW_CURL:-curl}" -V)"
+    if [[ "${curl_version}" =~ HTTP/2 ]]
+    then
+      output="${output} HTTP2"
+    fi
+  fi
```

Comment:
It might be easier and cheaper to just add an `append_to_cccfg` here: https://github.com/Homebrew/brew/blob/main/Library/Homebrew/extend/os/mac/extend/ENV/super.rb#L161 gated to >= 10.13 && < 14, using a letter that isn't already used like `h` or `C` or something. Then here you can check `if [[ "${HOMEBREW_CCCFG}" = *h* ]]` before running most of the code here.

Also makes it more obvious we can remove it if we drop < macOS 14 support in the far future and means none of the hacks will run in macOS 14 and later which is safer in case there's edge cases we've missed.

Reply:
make sense 👍